packages/mui-material/src/Fab/Fab.js

```diff
@@ -50,28 +50,30 @@ const FabRoot = styled(ButtonBase, {
     minWidth: 0,
     width: 56,
     height: 56,
-    zIndex: theme.zIndex.fab,
-    boxShadow: theme.shadows[6],
+    zIndex: (theme.vars || theme).zIndex.fab,
+    boxShadow: (theme.vars || theme).shadows[6],
     '&:active': {
-      boxShadow: theme.shadows[12],
+      boxShadow: (theme.vars || theme).shadows[12],
     },
-    color: theme.palette.getContrastText(theme.palette.grey[300]),
-    backgroundColor: theme.palette.grey[300],
+    color: theme.vars
+      ? theme.vars.palette.text.primary
```

Comment:
This looks good to me because `getContrastText(grey[300])` 99.99% returns `text.primary`. cc @mnajdova

Reply:
leaves 0.01% for the super edge cases where custom `grey[300]` has high contrast.

Reply:
Sounds good 👍 We will see if people complain (I don't expect it will happen :))